ros2launch/ros2launch/command/launch.py

```diff
@@ -49,17 +97,18 @@ def add_arguments(self, parser, cli_name):
         arg = parser.add_argument(
             'package_name',
             help='Name of the ROS package which contains the launch file')
-        arg.completer = package_name_completer
+        arg.completer = package_name_or_launch_file_completer
         arg = parser.add_argument(
             'launch_file_name',
             # TODO(wjwwood) make this not optional when full launch path is supported.
             nargs='?',
             help='Name of the launch file')
+        arg.completer = LaunchFileNameCompleter()
         arg = parser.add_argument(
             'launch_arguments',
             nargs='*',
             help="Arguments to the launch file; '<name>:=<value>' (for duplicates, last one wins)")
-        arg.completer = LaunchFileNameCompleter()
+        arg.completer = SuppressCompleterWorkaround()
```

Comment:
Why is `SuppressCompleter` not sufficient? It seems to work in other places, e.g. https://github.com/colcon/colcon-package-information/blob/8360ca881e0c51edd8696f3683029badcc301835/colcon_package_information/verb/list.py#L95-L100

Reply:
https://github.com/kislyuk/argcomplete/pull/289

Reply:
> Can you elaborate why it does work without the workaround in other cases but not here?

Using
```
$ tree
.
├── composition
│   ├── asdf.launch.py
│   ├── foo
│   │   ├── bar.launch.py
│   │   └── baz.launch.py
│   └── foo.launch.py
├── composition_launch.py
└── file_i_dont_want_completed.txt
```

With `SupressCompleter()`

```
$ ros2 launch composition <tab><tab>
composition/                    composition_launch.py           file_i_dont_want_completed.txt
```

With `SupressCompleterWorkaround()`

```
$ ros2 launch composition <tab><tab>
-a                              --debug                         --print-description             --show-args
composition_demo.launch.py      -p                              -s                              --show-arguments
-d                              --print                         --show-all-subprocesses-output 
```

I haven't tested `colcon-package-information`, so I don't know if `SuppressCompleter` is working as intended there.

Reply:
> Can you elaborate why it does work without the workaround in other cases but not here?

Using
```
$ tree
.
├── composition
│   ├── asdf.launch.py
│   ├── foo
│   │   ├── bar.launch.py
│   │   └── baz.launch.py
│   └── foo.launch.py
├── composition_launch.py
└── file_i_dont_want_completed.txt
```

With `SupressCompleter()`

```
$ ros2 launch composition <tab><tab>
composition/                    composition_launch.py           file_i_dont_want_completed.txt
```

With `SupressCompleterWorkaround()`

```
$ ros2 launch composition <tab><tab>
-a                              --debug                         --print-description             --show-args
composition_demo.launch.py      -p                              -s                              --show-arguments
-d                              --print                         --show-all-subprocesses-output 
```

I haven't tested `colcon-package-information`, so I don't know if `SuppressCompleter` is working as intended there.

Reply:
> I don't know if `SuppressCompleter` is working as intended there.

It does.

What version of `argcomplete` are you using?

Reply:
`argcomplete==1.11.1`, and I was also able to reproduce using upstream `master`

Reply:
Maybe I am missing the part why it is necessary in this case only.